krcal/core/kr_parevol_functions_test.py

```diff
@@ -0,0 +1,124 @@
+    mask_s1[masks_time[1]] = np.random.choice([True, False], sum(masks_time[1]), p=(probs_s1[1], 1-probs_s1[1]))
+    mask_s1[masks_time[2]] = np.random.choice([True, False], sum(masks_time[2]), p=(probs_s1[2], 1-probs_s1[2]))
+
+    mask_cut               = masks_container(s1   = mask_s1,
+                                             s2   = mask_s1,
+                                             band = mask_s1)
+    #the expected relative efficiencies for other cuts are 1 since the mask is repeated
+    pars_expected          = pd.DataFrame({'ts':ts,
+                                           'S1eff' : probs_s1,
+                                           'S2eff': [1.]*3,
+                                           'Bandeff':[1.]*3})
+    pars                   = pd.DataFrame({'ts'   : ts})
+    pars_out               = cut_time_evolution(masks_time, dst, mask_cut, pars)
+
+    assert_dataframes_close(pars_expected, pars_out, atol=5e-2)
+
+def test_cut_time_evolution_multievents(folder_test_dst,
+                                        test_dst_file ):
+    dst          = load_dst(folder_test_dst+test_dst_file, "DST", "Events").iloc[:1000]
+    dst_uniquevs = dst.drop_duplicates('event')
+    #make sure that the dataframe repeated 3x gives the same results as the original one
+    mask_S1_ue   = np.random.choice([True, False], len(dst_uniquevs))
+    dst_multievs = dst_uniquevs.append([dst_uniquevs]*2, ignore_index=True)
+    mask_S1_me   = np.tile(mask_S1_ue, 3)
+    mask_cut_ue  = masks_container(s1   = mask_S1_ue,
+                                   s2   = mask_S1_ue,
+                                   band = mask_S1_ue)
+    mask_cut_me  = masks_container(s1   = mask_S1_me,
+                                   s2   = mask_S1_me,
+                                   band = mask_S1_me)
+    min_time        = dst_uniquevs.time.min()
+    max_time        = dst_uniquevs.time.max()
+    ts_ue, masks_ue = get_time_series_df(time_bins  = 1,
+                                         time_range = (min_time, max_time),
+                                         dst        = dst_uniquevs)
+    ts_me, masks_me = get_time_series_df(time_bins  = 1,
+                                         time_range = (min_time, max_time),
+                                         dst        = dst_multievs)
+    pars_ue      = pd.DataFrame({'ts'   : ts_ue})
+    pars_me      = pd.DataFrame({'ts'   : ts_me})
+    pars_out_ue  = cut_time_evolution(masks_ue, dst_uniquevs, mask_cut_ue, pars_ue)
+    pars_out_me  = cut_time_evolution(masks_me, dst_multievs, mask_cut_me, pars_me)
+    assert_dataframes_close(pars_out_ue, pars_out_me)
+
+def test_cut_time_evolution_unique_events(folder_test_dst,
+                                          test_dst_file ):
+
+    dst             = load_dst(folder_test_dst + test_dst_file, "DST", "Events")
+    mask_S1         = dst         .nS1   == 1
+    mask_S2         = dst[mask_S1].nS2   == 1
```

Comment:
Iam a bit confused with the mask_S2. As it is defined now it will be pandas Series with smaller size than mask_S1. The reason why you could still do mask_S1 & mask_S2 is because pandas will keep False by default if the rows are not contained in both series, so technically is not wrong but is also not what we do in apply_cuts function. Also it would not work if mask_S1 and mask_S2 were numpy arrays. I dont think this tests brings anything that is not contained in the previous test, but if you think it is needed, fix the dimensions and aligning and im happy to approve this PR :)

Reply:
Yes, the mask was not intended to be a subset. I agree it does not add any check to the previous test, I will proceed to remove it.

Reply:
Should I remove and keep it in the commit line, or is it better to pretend it never existed?